daemon/config.go

```diff
@@ -75,6 +75,7 @@ func (config *Config) InstallFlags() {
 	config.Ulimits = make(map[string]*ulimit.Ulimit)
 	opts.UlimitMapVar(config.Ulimits, []string{"-default-ulimit"}, "Set default ulimits for containers")
 	flag.StringVar(&config.LogConfig.Type, []string{"-log-driver"}, "json-file", "Default driver for container logs")
+	opts.LogOptsVar(config.LogConfig.Config, []string{"-log-opt"}, "Set log driver options")
```

Comment:
@LK4D4 why not use `opts.NewListOpts` with opts.ValidateEnv like --label ?
Also there are already functions like convertKVStringsToMap that we can reuse.